src/main/java/jdraw/std/BasicGrid.java

```diff
@@ -0,0 +1,56 @@
+package jdraw.std;
+
+import jdraw.framework.DrawGrid;
+import jdraw.framework.Figure;
+import jdraw.framework.FigureHandle;
+
+import java.awt.*;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class BasicGrid implements DrawGrid {
+    public final int step = 50;
+    private boolean activated = false;
+
+    @Override
+    public Point constrainPoint(Point p) {
+        int yDelta = p.y % step;
+        int xDelta = p.x % step;
```

Comment:
Attention: if p.x or p.y is negative, then xDelta or yDelta is negative as well, i.e. the test whether xyDelta is > step/2 is always false, which is not the intended behavior. Negative positions on the other side are not usual, but if you spawn a new figure and move the mouse above the upper border and left to the left window border, then negative coordinates are mapped.